packages/react-native-editor/__device-tests__/gutenberg-editor-media-blocks-@canary.test.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Internal dependencies
+ */
+import { blockNames } from './pages/editor-page';
+import { waitForMediaLibrary, isAndroid } from './helpers/utils';
+import testData from './helpers/test-data';
+
+const onlyOniOS = ! isAndroid() ? describe : describe.skip;
+
+describe( 'Gutenberg Editor Audio Block tests', () => {
+	it( 'should be able to add an audio block and a file to it', async () => {
+		// add an audio block
+		await editorPage.addNewBlock( blockNames.audio );
+
+		// dismiss the media picker automatically opened when adding an audio block
+		await waitForMediaLibrary( editorPage.driver );
+		await editorPage.closePicker();
+
+		// verify there's an audio block
+		let block = await editorPage.getFirstBlockVisible();
+		await expect( block ).toBeTruthy();
+
+		// tap on the audio block
+		block.click();
+
+		// wait for the media picker's Media Library option to come up
+		await waitForMediaLibrary( editorPage.driver );
+
+		// tap on Media Library option
+		await editorPage.chooseMediaLibrary();
+
+		// get the html version of the content
+		const html = await editorPage.getHtmlContent();
+
+		// verify the html matches the expected content
+		expect( html.toLowerCase() ).toBe(
+			testData.audioBlockPlaceholder.toLowerCase()
+		);
+
+		block = await editorPage.getBlockAtPosition( blockNames.audio );
+		await block.click();
+		await editorPage.removeBlock();
+	} );
+} );
+
+describe( 'Gutenberg Editor File Block tests', () => {
+	it( 'should be able to add a file block and a file to it', async () => {
+		// add a file block
+		await editorPage.addNewBlock( blockNames.file );
+
+		// dismiss the media picker automatically opened when adding a file block
+		await waitForMediaLibrary( editorPage.driver );
+		await editorPage.closePicker();
+
+		// verify there's a file block
+		let block = await editorPage.getFirstBlockVisible();
+		await expect( block ).toBeTruthy();
+
+		// tap on the file block
+		block.click();
+
+		// wait for the media picker's Media Library option to come up
+		await waitForMediaLibrary( editorPage.driver );
+
+		// tap on Media Library option
+		await editorPage.chooseMediaLibrary();
+
+		// get the html version of the content
+		const html = await editorPage.getHtmlContent();
+
+		// verify the html matches the expected content
+		expect( html.toLowerCase() ).toBe(
+			testData.fileBlockPlaceholder.toLowerCase()
+		);
+
+		block = await editorPage.getBlockAtPosition( blockNames.file );
+		await block.click();
+		await editorPage.removeBlock();
+	} );
+} );
+
+// iOS only test - Can only add image from media library on iOS.
```

Comment:
```suggestion
// iOS only test - It can only add images from the media library on iOS.
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/48026/commits/bff19d2306ff8244cae576f4f11134029628a52e